### DISCUSSION TITLE
onlick ein Fenster öffnen mit dynamischen Content

### DISCUSSION THREAD

**Discussion by devchristiansimon** [neuefische/questions#12] [Web] [answered]
Heyho,
ich zerbreche mir den Kopf wie ich auf Click von einen meiner Karten ein Fenster öffne.
Ich hab mir dazu schon die [window.open](https://developer.mozilla.org/en-US/docs/Web/API/Window/open) - Funktion angeschaut, aber die benötigt eine URL. 
Ich möchte, dass dynamisch im Fenster Inhalte von meiner JSON Datei bezugnehmend zur Card, welche angeklickt wurde erscheint.

Das Fenster soll auch ein X für Schließen haben. Solche Art von Fenster habe ich schon auf diversen Webseiten gefunden, nur fehlt mir hier der Ansatz.

Macht es hier auch schon wegen der Accessibility Sinn den Content dynamisch in einer neuen Seite zu laden?

**Comment by actyvystom** [comment, marked as answer]
Hallo Gustl :wave:,

Du könntest das "Fenster" auch als Komponente implementieren, die über eine State-Variable umgeschaltet werden kann. Dieser Fenster-(/Modal)-Komponente kannst Du dann die Daten als Props übergeben, die Sichtbarkeit steuerst Du aus Deiner übergeordneten Komponente. Somit hast Du auch mit Blick auf Barrierefreiheit kein Problem, Deine Fenster/Modal-Komponente kann dann auch A11y-konformes HTML rendern.
Sehr grob skizziert:

```javascript
...
const [modalVisibility, setModalVisibility] = useState(false);
...
<ModalWindow data={myData} visible={modalVisibility}/>
...
```


Ich hoffe, diese Hinweise helfen Dir weiter.

**Reply by stefankorder** [reply]
So könnte man es lösen 👍🏻

Hinter `visible` versteckt sich ein `boolean` der dann dazu genutzt werden kann um darüber zu Entscheiden, ob ein Element (In diesem Fall das Modal) die CSS-Eigenschaft `display: none;` hat oder nicht.
